packages/babel-parser/src/parser/expression.js

```diff
@@ -752,9 +754,10 @@ export default class ExpressionParser extends LValParser {
   }
 
   finishOptionalCallExpression(node: N.CallExpression): N.CallExpression {
-    if (node.callee.type === "Import") {
+    validate: if (node.callee.type === "Import") {
       if (node.arguments.length !== 1) {
         this.raise(node.start, "import() requires exactly one argument");
+        break validate;
```

Comment:
TODO: Use if/else instead.

Reply:
@nicolo-ribaudo It's an interesting PR, and you are doing a great work! But it will be difficult to recover from all errors. E.g. block statements and other places where you normally only do a token EOF check. And also in the formal params. It quickly runs into a infinity loop. 
TS have a solution for block statement - 2 huge switch statements to figure out statement and expressions. Maybe an idea? 

In my experimental JS / TS parser I'm simply using this code now to prevent infinity loops
 `while (parser.token & Token.IsStatementStart) {}`. 

Anyway! Keep up the good work!